Sandbox/BubbleSortLayer.cpp

```diff
@@ -28,31 +28,34 @@ void BubbleSortLayer::onEvent(const Aquarius::Event& event)
 {
     auto keyPressEvent = static_cast<const Aquarius::KeyPressedEvent &>(event);
 
-    switch (keyPressEvent.getCode()) {
-        case (Aquarius::Input::KeyCode::Key_up): {
-            if (delay >= 0) {
-                delay = delay - 200;
-            }
+    switch (keyPressEvent.getCode())
+    {
+        case (Aquarius::Input::KeyCode::Key_r):
+            {
+            resetSort = true;
             break;
-        }
-
-        case (Aquarius::Input::KeyCode::Key_down): {
-            if (delay >= 0) {
-                delay = delay + 200;
             }
 
-            break;
-        }
+        case (Aquarius::Input::KeyCode::Key_p):
+            {
+            if(!pauseSort)
+            {
+                pauseSort = true;
+            }
 
-        case (Aquarius::Input::KeyCode::Key_r): {
-            resetSort = true;
+            else
+            {
+                pauseSort = false;
+            }
             break;
-        }
+            }
     }
 }
 
-void BubbleSortLayer::onUpdate(Aquarius::timeDelta_t ts)
+void BubbleSortLayer::onUpdate(Aquarius::timeDelta_t dt)
 {
+    delay = dt * 750;
```

Comment:
You could make `delay` equal to 750 in the class definition similar to what you did before, and then create a local variable that is equal to dt * delay and use that when you pause the thread. This might make it easier to change the delay at runtime if you end up doing something like that through imGUI

Reply:
I second this, it would be very nice to allow this to be changed via ImGUI